src/containers/PipelineRun/PipelineRun.js

```diff
@@ -188,166 +75,18 @@ export /* istanbul ignore next */ class PipelineRunContainer extends Component {
   }
 
   render() {
-    const { match, error, intl } = this.props;
-    const { pipelineRunName } = match.params;
-
-    const {
-      selectedStepId,
-      selectedTaskId,
-      loading,
-      showRebuildNotification
-    } = this.state;
-
-    if (loading) {
-      return <StructuredListSkeleton border />;
-    }
-
-    if (error) {
-      return (
-        <InlineNotification
-          kind="error"
-          hideCloseButton
-          lowContrast
-          title={intl.formatMessage({
-            id: 'dashboard.pipelineRun.error',
-            defaultMessage: 'Error loading PipelineRun'
-          })}
-          subtitle={getErrorMessage(error)}
-        />
-      );
-    }
-
-    if (!this.props.pipelineRun) {
-      return (
-        <InlineNotification
-          kind="info"
-          hideCloseButton
-          lowContrast
-          title="Cannot load PipelineRun"
-          subtitle={`PipelineRun ${pipelineRunName} not found`}
-        />
-      );
-    }
-
-    const {
-      error: pipelineRunError,
-      pipelineRun,
-      taskRunNames
-    } = this.loadPipelineRunData();
-
-    const {
-      lastTransitionTime,
-      reason: pipelineRunReason,
-      status: pipelineRunStatus
-    } = getStatus(pipelineRun);
-
-    if (pipelineRunError) {
-      return (
-        <>
-          <RunHeader
-            lastTransitionTime={lastTransitionTime}
-            loading={loading}
-            pipelineRun={pipelineRun}
-            runName={pipelineRunName}
-            reason="Error"
-            status={pipelineRunStatus}
-          />
-          <InlineNotification
-            kind="error"
-            hideCloseButton
-            lowContrast
-            title={`Unable to load PipelineRun details: ${
-              pipelineRunError.reason
-            }`}
-            subtitle={pipelineRunError.message}
-          />
-        </>
-      );
-    }
-    const taskRuns = this.loadTaskRuns(pipelineRun, taskRunNames);
-    const taskRun = selectedTaskRun(selectedTaskId, taskRuns) || {};
-
-    const { definition, reason, status, stepName, stepStatus } = taskRunStep(
-      selectedStepId,
-      taskRun
-    );
-
-    const logContainer = (
-      <Log
-        fetchLogs={() => fetchLogs(stepName, stepStatus, taskRun)}
-        key={stepName}
-        stepStatus={stepStatus}
-      />
-    );
+    const { error, pipelineRun, tasks, taskRuns } = this.props;
+    const { loading } = this.state;
 
     return (
-      <>
-        {showRebuildNotification && !showRebuildNotification.logsURL && (
```

Comment:
Since we want consumers to be able to omit / customise the Rebuild button and its functionality, wouldn't it make more sense to keep the rebuild notification and related code here in the container as it's dashboard specific? The published component could then be simplified to remove all of that code.

Reply:
Was going to do it seperately, Might not be trivial, need to sort out how we handle notifications